src/main/java/mSearch/filmlisten/FilmlisteLesen.java

```diff
@@ -119,125 +82,187 @@ private InputStream selectDecompressor(String source, InputStream in) throws Exc
+                    //if we are in FASTAUTO mode, we don´t need film descriptions.
+                    //this should speed up loading on low end devices...
+                    if (workMode == WorkMode.FASTAUTO) {
+                        if (DatenFilm.JSON_NAMES[i] == DatenFilm.FILM_BESCHREIBUNG
+                                || DatenFilm.JSON_NAMES[i] == DatenFilm.FILM_WEBSEITE
+                                || DatenFilm.JSON_NAMES[i] == DatenFilm.FILM_GEO) {
+                            jp.nextToken();
+                            continue;
                         }
                     }
-                    if (datenFilm.arr[DatenFilm.FILM_SENDER].isEmpty()) {
-                        datenFilm.arr[DatenFilm.FILM_SENDER] = sender;
+                    if (DatenFilm.JSON_NAMES[i] == DatenFilm.FILM_NEU) {
+                        final String value = jp.nextTextValue();
+                        //This value is unused...
+                        //datenFilm.arr[DatenFilm.FILM_NEU_NR] = value;
+                        datenFilm.setNew(Boolean.parseBoolean(value));
                     } else {
-                        sender = datenFilm.arr[DatenFilm.FILM_SENDER];
+                        datenFilm.arr[DatenFilm.JSON_NAMES[i]] = jp.nextTextValue();
                     }
-                    if (datenFilm.arr[DatenFilm.FILM_THEMA].isEmpty()) {
-                        datenFilm.arr[DatenFilm.FILM_THEMA] = thema;
-                    } else {
-                        thema = datenFilm.arr[DatenFilm.FILM_THEMA];
+
+                    /// für die Entwicklungszeit
+                    if (datenFilm.arr[DatenFilm.JSON_NAMES[i]] == null) {
+                        datenFilm.arr[DatenFilm.JSON_NAMES[i]] = "";
                     }
+                }
+                if (datenFilm.arr[DatenFilm.FILM_SENDER].isEmpty()) {
+                    datenFilm.arr[DatenFilm.FILM_SENDER] = sender;
+                } else {
+                    sender = datenFilm.arr[DatenFilm.FILM_SENDER];
+                }
+                if (datenFilm.arr[DatenFilm.FILM_THEMA].isEmpty()) {
+                    datenFilm.arr[DatenFilm.FILM_THEMA] = thema;
+                } else {
+                    thema = datenFilm.arr[DatenFilm.FILM_THEMA];
+                }
 
-                    listeFilme.importFilmliste(datenFilm);
-                    if (seconds > 0) {
-                        // muss "rückwärts" laufen, da das Datum sonst 2x gebaut werden muss
-                        // wenns drin bleibt, kann mans noch ändern
-                        if (!checkDate(datenFilm)) {
-                            listeFilme.remove(datenFilm);
-                        }
+                listeFilme.importFilmliste(datenFilm);
+                if (milliseconds > 0) {
+                    // muss "rückwärts" laufen, da das Datum sonst 2x gebaut werden muss
+                    // wenns drin bleibt, kann mans noch ändern
+                    if (!checkDate(datenFilm)) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS134)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1066)